vmm/src/vm.rs

```diff
@@ -968,6 +987,29 @@ impl Vm {
         Ok(EntryPoint { entry_addr })
     }
 
+    #[cfg(feature = "igvm")]
+    fn load_igvm(
+        igvm: File,
+        memory_manager: Arc<Mutex<MemoryManager>>,
+        cpu_manager: Arc<Mutex<cpu::CpuManager>>,
+    ) -> Result<EntryPoint> {
+        /*
+        BIOS-e820: [mem 0x0000000000000000-0x000000000009ffff] usable
+        BIOS-e820: [mem 0x00000000000a0000-0x00000000000fffff] reserved
+        BIOS-e820: [mem 0x0000000000100000-0x00000000001fffff] ACPI data
+        BIOS-e820: [mem 0x0000000001a00000-0x0000000003c79fff] usable
+         */
+        let arch_mem_regions: Vec<ArchMemRegion> = Vec::new();
+        //TODO Generate the memory layout in arch/x86/layout.rs
+
+        let res = igvm_loader::load_igvm(&igvm, memory_manager, cpu_manager, arch_mem_regions, "")
```

Comment:
I agree. The last commit needs quite some cleanup since we decided to keep the layout open for future work. Also, please add the a reference to the issue as a part of the `TODO:` comments.